### PR TITLE
Before enqueueing the job register it with the exact time it was supposed to run

### LINKED ISSUE #181: Use rufus job's schedule time to determine if a specific job instance has ran before
Job's schedule time is the timestamp at which a rufus job instance was scheduled to be triggered. 

rufus-scheduler gets `EoTime.now` and then triggers each job passing that time to `#trigger` method. 

This could cause problems if  `Rufus::Scheduler#trigger_jobs` is invoked a second after a particular jobs has been triggered.

### PR DESCRIPTION
**Pull request by marcelolx**
TL;DR; Work extracted from #276, thanks @afrase!

#156 had some interesting discussion around this and why it is more important to fix this first before trying to introduce locking which not necessarily would avoid double execution of scheduled jobs.

# Details

Rufus calls the block with the current time instead of the time the job is scheduled to run. If you have multiple hosts, it's possible
that Rufus will run the block at slightly different times which results in a job getting run multiple times.

This figures out when the job was scheduled to run instead of the time Rufus sends.

Co-Authored-By: Aaron Frase <afrase@users.noreply.github.com>

TODO:
- [ ] Tests
- [ ] Detail commit message/PR description
- [ ] Check if README needs to be updated

Closes #181
Closes #276

This may solve #332 too, but I need to think more about that scenario (at least A).

### CODE REVIEW

**Comment by bhaskar-simpl** [comment]
@marcelolx Do you need help closing and releasing this?

**Comment by marcelolx** [comment]
@bhaskar-simpl I have had no time lately to write some tests for this and test it, so any help is welcome; feel free to send a PR against the branch of this one or master.

**Comment by marcelolx** [comment]
Closing in favor of #463
